### PR TITLE
Fix logging

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/log/SonarLintLogging.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    LevelFilter levelFilter = new LevelFilter();
+    levelFilter.setLevel(Level.ERROR);
+    defaultAppender.addFilter(levelFilter);
+
+    // if output is not set for thread, keep default behavior (configured to stdout)
+    setCustomRootAppender(defaultAppender);
+  }
+
+  private static void setCustomRootAppender(Appender<ILoggingEvent> defaultAppender) {
+    Logger logger = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/log/SonarLintLogging.java]: MAJOR Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1312)

File: core/src/main/java/org/sonarsource/sonarlint/core/log/SonarLintLogging.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    logger.detachAndStopAllAppenders();
+
+    appender = new LogCallbackAppender(defaultAppender);
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/log/SonarLintLogging.java]: CRITICAL Synchronize this lazy initialization of 'appender' [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2444)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* CRITICAL 1 critical
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* CRITICAL 1 critical
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: core/src/main/java/org/sonarsource/sonarlint/core/log/SonarLintLogging.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    defaultAppender.addFilter(levelFilter);
+
+    // if output is not set for thread, keep default behavior (configured to stdout)
+    setCustomRootAppender(defaultAppender);
+  }
+
+  private static void setCustomRootAppender(Appender<ILoggingEvent> defaultAppender) {
+    Logger logger = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
+
+    appender = new LogCallbackAppender(defaultAppender);
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/log/SonarLintLogging.java]: CRITICAL Synchronize this lazy initialization of 'appender' [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2444)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* CRITICAL 1 critical
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* CRITICAL 1 critical
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: core/src/main/java/org/sonarsource/sonarlint/core/log/SonarLintLogging.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.log;
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/log/SonarLintLogging.java]: Is there a way to put that in the API? If not you should be careful to export the package in MANIFEST for OSGi (see pom.xml)

**Pull request merged**